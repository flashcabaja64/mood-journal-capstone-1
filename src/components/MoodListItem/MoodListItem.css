.mood-entry {
  width: 100%;
  margin: 1rem;
  line-height: 1.5rem;
  text-align: center;
}

.wrap-mood-entry {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  min-height: 10vh;
  position: relative;
  z-index: 1;
  border-radius: 27px;
}

.entry-wrap {
  border-radius: 27px;
  overflow: hidden;
  width: 30rem;
  position: relative;
  padding: 15px 0 20px 0;
  box-shadow: 0 3px 20px 0px rgba(255, 255, 255, 0.089)
}
  /*                  */
 /*                     */
/*                        */
/* Original [Focus Input (class)] in LoginForm.css */
/*                      */
/*                     */
/*                    */

/* Entry details */
.entry-title {
  font-size: 2.1rem;
  font-weight: 700;
  margin: 1% 0 4% 0;
  line-height: 1.2;
}

.entry-content {
  font-size: 1rem;
  margin: 1% 0 7% 0;
  padding: 0 2% 0 2%;
}

.entry-duration, .entry-type {
  font-size: .8rem;
}

.entry-btn-container {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  padding-top: 10px;
}

.edit-btn, .del-btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  padding: 0 10px;
  width: 30%;
  height: 30px;
  background-color: rgb(0, 29, 54);
  border-radius: 25px;
  border: none;
  cursor: pointer;
  text-decoration: none;

  font-family: 'Arial';
  letter-spacing: 2px;
  line-height: 1.2;
  color: white;
  font-size: 16px;
  text-transform: uppercase;
}

.edit-btn {
  width: 200%;
}

.edit-btn-link {
  width: 15%;
}

button.edit-btn:hover, button.del-btn:hover {
  background-color: #1A5486;
}

@-webkit-keyframes btn-animation {
  to {
    -webkit-transform: scale(1.1);
    transform: scale(1.1);
  }
}
@keyframes btn-animation {
  to {
    -webkit-transform: scale(1.1);
    transform: scale(1.1);
  }
}
.del-btn, .edit-btn {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
}
.del-btn:hover, .del-btn:focus, .del-btn:active,
.edit-btn:hover, .edit-btn:focus, .edit-btn:active {
  -webkit-animation-name: btn-animation;
  animation-name: btn-animation;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}