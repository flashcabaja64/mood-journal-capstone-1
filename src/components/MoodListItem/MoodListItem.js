import React, { Component } from 'react';
import { Link } from 'react-router-dom';
//import config from '../../config'
import MoodContext from '../../MoodContext/MoodContext';
import EntryApiService from '../../services/entry-api-service';
import './MoodListItem.css'
//import EditForm from '../EditForm/EditForm'

export default class MoodListItem extends Component {

	static contextType = MoodContext

	//work on this later (error: SyntaxError: Unexpected end of JSON input)
	handleDeleteClick = e => {
		e.preventDefault()
		const entryId = this.props.id;
		const { deleteEntryPage } = this.context;

		EntryApiService.deleteEntry(entryId)
			.then(() => {
				console.log('working delete')
				deleteEntryPage(entryId)
			})
			.catch(err => {
				console.error(err)
			})
	}

	render() {
		const { entry } = this.props
		return (
				<div className="mood-entry">
					<div className="wrap-mood-entry">
						<div className="entry-wrap">
							<h1 className="entry-title">{entry.title}</h1>
							<span className="focus-input"></span>
							<p className="entry-content">{entry.content}</p>
							<span className="focus-input"></span>
							<p className="entry-duration">Mood duration: {entry.duration} hours</p>
							<span className="focus-input"></span>
							<p className="entry-type">Mood Type: {entry.mood_type}</p>
							<span className="focus-input"></span>
							<div className="entry-btn-container">
								<Link to={`/entries/${entry.id}`} className="edit-btn-link">
									<button
										type="button"
										className="edit-btn"
									>
										Edit
									</button>
								</Link>
								<span className="focus-input"></span>
								<button
									type='button'
									className="del-btn"
									onClick={this.handleDeleteClick}
								>
									Delete
								</button>
								<span className="focus-input"></span>
							</div>
						</div>
					</div>
				</div>
		)
	}
}