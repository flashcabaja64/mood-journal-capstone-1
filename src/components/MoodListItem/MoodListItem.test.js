import React from 'react';
import ReactDOM from 'react-dom';
import MoodListItem from './MoodListItem';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('MoodListItem component testing', () => {
  const entry = {
    id: 1,
    title: 'title 1',
    content: 'content 1',
    duration: 1,
    mood_type: 'Low'
  }
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(
    <MemoryRouter>
      <MoodListItem entry={entry} index={1} />
    </MemoryRouter>
    , div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(
          <MemoryRouter>
            <MoodListItem entry={entry} index={1}/>
          </MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })
})