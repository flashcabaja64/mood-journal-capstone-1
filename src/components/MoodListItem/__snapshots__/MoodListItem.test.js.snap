// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodListItem component testing renders UI as expected 1`] = `
<div
  className="mood-entry"
>
  <div
    className="wrap-mood-entry"
  >
    <div
      className="entry-wrap"
    >
      <h1
        className="entry-title"
      >
        title 1
      </h1>
      <span
        className="focus-input"
      />
      <p
        className="entry-content"
      >
        content 1
      </p>
      <span
        className="focus-input"
      />
      <p>
        Mood duration: 
        1
         hours
      </p>
      <span
        className="focus-input"
      />
      <p>
        Mood Type: 
        Low
      </p>
      <span
        className="focus-input"
      />
      <div
        className="entry-btn-container"
      >
        <a
          className="edit-btn-link"
          href="/entries/1"
          onClick={[Function]}
        >
          <button
            className="edit-btn"
            type="button"
          >
            Edit
          </button>
        </a>
        <span
          className="focus-input"
        />
        <button
          className="del-btn"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
        <span
          className="focus-input"
        />
      </div>
    </div>
  </div>
</div>
`;
